brave/brave-core
# WIP: Rewards data can now be cleared

Fixes brave/brave-browser#926

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan 1:

* Start Brave - Clean profile - Do not enable rewards
* Go to settings, then clear rewards data
* Check "Clear All Rewards" and make sure "Clear Rewards Auto-Contribute Sites" is grayed out.
* Uncheck "Clear All Rewards" and make sure "Clear Rewards Auto-Contribute Sites" is available.
* Clear Auto-Contribute Sites and make sure the browser does not crash.
* Reopen Clear Rewards Data and check Clear All Rewards and press clear data.
* Make sure you get a warning about Rewards data loss.
* Press clear data and make sure browser does not crash

## Test Plan 2:

* Start Brave - Clean profile - Enable rewards
* Visit a site for the required amount of time to place in auto-contribute list
* Go to settings, then clear rewards data
* Observe that Auto-Contribute Sites option shows "1 site"
* Clear Auto-Contribute Sites
* Open Rewards panel and make sure that auto-contribute list is empty

## Test Plan 3:

* Start Brave - Clean profile - Enable rewards
* Visit a few sites for the required amount of time to place in auto-contribute list
* Go to settings, then clear rewards data
* Observe that Auto-Contribute Sites option shows "(#) sites"
* Clear Auto-Contribute Sites
* Open Rewards panel and make sure that auto-contribute list is empty

## Test Plan 4:

* Start Brave with a reconcile interval - Clean profile - Enable rewards
* Visit a few sites for the required amount of time to place in auto-contribute list
* Go to settings, then clear rewards data
* When contribution starts, make sure checkboxes are grayed out and that you see a message about not being able to clear during contribution
* close clear rewards
* reopen. Make sure text under auto-contribute says 'None since your last contribution on [date of contribution]'

## Test Plan 5:

* Start Brave - Clean profile - Enable rewards
* Visit a few sites for the required amount of time to place in auto-contribute list
* Go to settings, then clear rewards data
* Select "Clear All Rewards Data"
* When the warning pops up, click "Cancel"
* Go to Rewards and make sure Rewards data was not cleared.

## Test Plan 6:

* Start Brave - Clean profile - Enable rewards
* Visit a few sites for the required amount of time to place in auto-contribute list
* Visit some sites and make some tips
* Go to settings, then clear rewards data
* Select "Clear All Rewards Data" and clear the data
* Go to Rewards and make sure all data in auto-contribute and tips is cleared
* Make sure the correct amount of BAT is still retained

————————
## Additional Info

* When clearing only Auto-Contribute data, only the sites listed in the Auto-Contribute table (including the ones listed under "See All XX Sites" will be cleared. Auto-Contribute data from previous months is not affected and no other data that is visible to users is affected.

* Auto-Contribute data for previous months is untouched if a user selects to only "Clear Auto-Contribute Data". Auto-Contribute data for previous months is discarded if a user selects to "Clear All Rewards Data".

* BAT designated for unverified publishers through tipping is not affected by "Clear Auto-Contribute Data". BAT designated for unverified publishers through tipping WILL be discarded if "Clear All Rewards Data" is selected.

* Tips will be discarded whether recurring or one time if "Clear All Rewards" data is selected. Tips are not affected by "Clear Auto-Contribute"

* Past Auto-Contribute sites are currently not visible in the UI and past Auto-Contribute sites that didn't make vote count would also not be visible in the UI and would be discarded when "Clear All Rewards Data" is selected.

* Clearing any Rewards data only affects data available in publisher_info_db. The amount of BAT in a user's wallet is not affected in any way

## Reviewer Checklist:
